tests/cluster.go

```diff
@@ -83,7 +85,16 @@ var zapLogOnce sync.Once
 
 // NewTestServer creates a new TestServer.
 func NewTestServer(ctx context.Context, cfg *config.Config, services []string) (*TestServer, error) {
-	//  disable the heartbeat async runner in test
+	// use temp dir to ensure test isolation.
+	if cfg.DataDir == "" || strings.HasPrefix(cfg.DataDir, "default.") {
+		dirPattern := fmt.Sprintf("pd-server-test-%s-", cfg.Name)
+		tempDir, err := os.MkdirTemp("", dirPattern)
```

Comment:
What if we have already set the data dir in the config parameter?

Reply:
So we only use temp dir in this func when data dir is empty or default prefix